py_gtfs_rt_ingestion/lib/s3_utils.py

```diff
@@ -152,7 +151,10 @@ def move_s3_objects(files: List[str], to_bucket: str) -> None:
 
     # this is the default pool size for a ThreadPoolExecutor as of py3.8
     cpu_count = cast(int, os.cpu_count() if os.cpu_count() is not None else 1)
-    pool_size = min(32, cpu_count + 4)
+    # make sure each pool will have at least 50 files to move
+    files_per_pool = 50
+    max_pool_size = max(1, int(len(files) / files_per_pool))
+    pool_size = min(32, cpu_count + 4, max_pool_size)
```

Comment:
This limits are `move_s3_objects` pool size if we're not moving many files. Makes sure each pool with have at least 50 files to move, kind of an arbitrary number.